custom/icds_reports/management/commands/fix_missing_thr_forms.py

```diff
@@ -0,0 +1,118 @@
+from datetime import datetime
+
+from dateutil.relativedelta import relativedelta
+from django.core.management import BaseCommand
+from django.db import connections
+
+from corehq.apps.locations.models import SQLLocation
+from custom.icds_reports.const import (
+    DASHBOARD_DOMAIN
+)
+from custom.icds_reports.tasks import (
+    icds_state_aggregation_task
+)
+
+query_text_1 = """
+    UPDATE "agg_awc_{month}_5" t
+    SET thr_distribution_image_count = ut.thr_distribution_image_count
+    FROM  "icds_dashboard_thr_v2" ut
+    WHERE t.awc_id = ut.awc_id AND t.month = ut.month AND ut.month='{month}'
+"""
+
+query_text_2 = """
+    UPDATE "agg_awc_{month}_4" t
+    SET thr_distribution_image_count = ut.thr_distribution_image_count
+    FROM (
+        SELECT
+            supervisor_id,
+            SUM(thr_distribution_image_count) as thr_distribution_image_count
+        FROM "agg_awc_{month}_5"
+        GROUP BY state_id, district_id,block_id,supervisor_id
+    ) ut
+    WHERE t.supervisor_id = ut.supervisor_id
+"""
+
+query_text_3 = """
+    UPDATE "agg_awc_{month}_3" t
+    SET thr_distribution_image_count = ut.thr_distribution_image_count
+    FROM (
+        SELECT
+            block_id,
+            SUM(thr_distribution_image_count) as thr_distribution_image_count
+        FROM "agg_awc_{month}_4"
+        GROUP BY state_id, district_id,block_id
+    ) ut
+    WHERE t.block_id = ut.block_id
+"""
+
+query_text_4 = """
+    UPDATE "agg_awc_{month}_2" t
+    SET thr_distribution_image_count = ut.thr_distribution_image_count
+    FROM (
+        SELECT
+            district_id,
+            SUM(thr_distribution_image_count) as thr_distribution_image_count
+        FROM "agg_awc_{month}_3"
+        GROUP BY state_id, district_id
+    ) ut
+    WHERE t.district_id = ut.district_id
+"""
+
+query_text_5 = """
+    UPDATE "agg_awc_{month}_1" t
+    SET thr_distribution_image_count = ut.thr_distribution_image_count
+    FROM (
+        SELECT
+            state_id,
+            SUM(thr_distribution_image_count) as thr_distribution_image_count
+        FROM "agg_awc_{month}_2"
+        GROUP BY state_id
+    ) ut
+    WHERE t.state_id = ut.state_id
+"""
+
+
+class Command(BaseCommand):
+    help = "Fills the Missing THR Forms"
+
+    def add_arguments(self, parser):
+        parser.add_argument('date', type=str, nargs='?')
+
+    def handle(self, *args, **options):
+        date = options["date"] if options["date"] else "2019-09-01"
+        date = datetime.strptime(date, '%Y-%m-%d')
+        self.run_task(date)
+
+    def run_task(self, initial_date):
+        final_date = datetime.now()
+        final_date = final_date.replace(day=1)
+
+        # get the number of month intervals between the dates
+        intervals = final_date.month - initial_date.month + 12 * (final_date.year - initial_date.year) + 1
```

Comment:
I have mentioned this in other PRs but I think this method is hard to reason about and overcomplicates what is really a simple operation. This set of things can mostly be reduced to
```
date = start_date
while date < end_date:
    update_data(date)
    date = date + relativedelta(months=1)
```